yb-voyager/cmd/importData.go

```diff
@@ -684,6 +699,60 @@ func importData(importFileTasks []*ImportFileTask) {
 
 }
 
+func importTasksViaTaskPicker(pendingTasks []*ImportFileTask, state *ImportDataState, progressReporter *ImportDataProgressReporter, poolSize int) error {
+	// The code can produce `poolSize` number of batches at a time. But, it can consume only
+	// `parallelism` number of batches at a time.
+	batchImportPool = pool.New().WithMaxGoroutines(poolSize)
+	log.Infof("created batch import pool of size: %d", poolSize)
+
+	taskPicker, err := NewSequentialTaskPicker(pendingTasks, state)
```

Comment:
we are passing pendingTasks to the constructor here, but in the constructor we are still segregating pending and done tasks

Reply:
True, yeah, it's not really necessary to filter out for done tasks in the constructor.